frontends/concrete-python/examples/floating_point/floating_point.ipynb

```diff
@@ -0,0 +1,609 @@
+    "        print(f\"{random_sample=}\")\n",
+    "    \n",
+    "    # Scale it and round\n",
+    "    scaled_sample = (round(random_sample[0] * scaling_factor).astype(np.uint32), round(random_sample[1] * scaling_factor).astype(np.uint32))\n",
+    "    if verbose:\n",
+    "        print(f\"{scaled_sample=}\")\n",
+    "    \n",
+    "    # Encrypt\n",
+    "    encrypted_scaled_sample = circuit.encrypt(*scaled_sample)\n",
+    "    \n",
+    "    # Computations in FHE\n",
+    "    encrypted_scaled_result = circuit.run(*encrypted_scaled_sample)\n",
+    "    \n",
+    "    # Decrypt\n",
+    "    scaled_result = circuit.decrypt(encrypted_scaled_result)\n",
+    "    if verbose:\n",
+    "        print(f\"{scaled_result=}\")\n",
+    "    \n",
+    "    # Unscale\n",
+    "    result = scaled_result * 1. / scaling_factor\n",
+    "\n",
+    "    bounded_error = 2. / scaling_factor\n",
+    "    clear_result = np.sqrt(random_sample[0]**2 + random_sample[1]**2)\n",
+    "    \n",
+    "    if verbose:\n",
+    "        print()\n",
+    "        print(f\"FHE:   {result}\")\n",
+    "        print(f\"clear: {clear_result}\")\n",
+    "        print(f\"bounded error: {bounded_error}\")\n",
+    "\n",
+    "    are_almost_the_same(result, clear_result, bounded_error)\n",
+    "\n",
+    "    verbose = False\n",
+    "    \n",
+    "print(f\"\\nCompilation and test look good, with FHE execution time of about {(time() - time_begin) / nb_test_samples:.2f} seconds per inference\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "560956ce-8791-4dd8-b7c0-c9c489f22063",
+   "metadata": {},
+   "source": [
+    "We can remark that here, the function is much slower, since it includes PBS. Furthermore, the input of the PBS is as large as twice the square of typical inputs, which can quickly become large. That's why we have had to reduce the scaling factor and the precision. When more precision is needed, one would have to be a bit more patient."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "85995459-8036-4b18-92f0-261d524b1a6e",
+   "metadata": {},
+   "source": [
+    "## Final example"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "afc5c7eb-db95-49c7-accc-5897e53db414",
+   "metadata": {},
+   "source": [
+    "We finish with a more complex example, where one needs to understand what kind of computations are done and cancel some scaling factors during the computation, to ensure correct computations.\n",
+    "\n",
+    "Let's suppose we want to convert `f(x, y) = (x**2 + 4y) / 1.33` to FHE. What we'll do is that, we'll scale `x` and `y` with a scaling factor `scaling_factor`, before encryption. Then, in FHE, `x**2` will somehow have `scaling_factor**2` as a scaler while `4y` will only have `scaling_factor`: so, to keep the addition homogeneous, we need to multiply `4y` by `scaling_factor` in the FHE circuit. Remark we could also have divided `x**2` by `scaling_factor` to keep things homogeneous. \n",
```

Comment:
I think we can improve this line:
```
`x**2` will somehow have `scaling_factor**2`
```
It's not `somehow`, it's because during multiplication, scaling factors would be multiplied as well.

Reply:
Let me put

> Then, in `x**2`, scaling factors will multiply (so we'll have a `scaling_factor**2` scaler) while in `4y` we'll only have a scaler `scaling_factor`

Is it better?

Reply:
How about this?
```
Then, in x**2, scaling factors will multiply (so we'll have a the scaling factor of `scaling_factor**2`) while in 4y scaling factor would remain as `scaling_factor`.
```